src/shapes/Object/InteractiveObject.ts

```diff
@@ -414,7 +414,12 @@ export class InteractiveFabricObject<
     ctx.save();
     ctx.translate(options.translateX, options.translateY);
     ctx.lineWidth = 1 * this.borderScaleFactor;
-    if (!this.group) {
+    // since interactive groups have been introduced, an object could be inside a group and needing controls
+    // the following equality check `this.group === this.parent` covers:
+    // object without a group ( undefined === undefined )
+    // object inside a group
+    // excludes object inside a group but multi selected since group and parent will differ in value
```

Comment:
@zhe-he for change like this where the condition covers many things but doesn't say any explicitly is better to add an extensive comment.
Those things are easy to forget

Reply:
@asturur as you mentioned objects under active selection never get selected so is this needed?
Can we just drop the condition altogether?

Reply:
Is `isMoving` enough?

Reply:
The first iteration of @zhe-he PR indeed removed the condition, and that sounded strange to me.
He came back and he added back a different condition i don't know which is the edge case that would break.

It could be that testing 'this.isMoving' is enough, since if you are inside an active selection you are not moving but the activeSelection is.

I do not know it could also be that the opacity bug is added by this new condition.
I have to chec at this point

Reply:
his message explains it
https://github.com/fabricjs/fabric.js/pull/9735#issuecomment-2008944249

the active selection is going to trigger the children renderControl.
So:
- isMoving is not enough ( children do not move )
- borderOpacityWhenMoving is set by the active selection and not by the object

And while this last one sentence is debatable ( who controls the borderOpacityWhenMoving ? the active selection or the object ) i prefer to just don't touch it.
Seems a dumb feature overall, why the controls should go semi transparent ONLY when moving ( or when transforming ) ?

This for me would be trashed all together an developers that want something like this can use before:transform and after:trasnform event and change the color of the controls to something that has opacity.

Reply:
We can fix it since the work is done, but is definitely a candidate for removal